ozaria/site/templates/play/tome/spell-palette-view.jade

```diff
@@ -1,43 +1,15 @@
 mixin header(label, name)
-  - var translatedName = translate('apis.' + _.string.slugify(name).replace('-', '_'));
   a.section-header.btn.btn-small.btn-illustrated.btn-warning(role="button", id="#palette-header-" + _.string.slugify(label), data-panel="#palette-tab-" + _.string.slugify(label))
-    if name.toLocaleLowerCase() == translatedName.toLocaleLowerCase() || ['Methods', 'Events', 'Spawnable'].indexOf(name) != -1
-      span= translatedName
-    else
-      span= name + ' (' + translatedName + ')'
-    if label == 'api-area' || label == 'stuff-area'
-      div.glyphicon.glyphicon-chevron-down(style="float: right")
-    else
-      div.glyphicon.glyphicon-chevron-right(style="float: right")
+    span= name
 div.container
   div.left.nano
     div.nano-content.panel-group#spell-palette-api-bar
-      if view.level.get('type') == 'web-dev'
-        +header('api-area', 'HTML')
-      else
-        +header('api-area', 'Methods')
-      div.apis.panel-collapse.collapse.in#palette-tab-api-area
-        .properties.properties-this
-      if tabs
-        each entries, tab in tabs
-          +header(tab, tab)
-          div.panel-collapse.collapse(id="palette-tab-" + _.string.slugify(tab))
-            div(class="properties properties-" + _.string.slugify(tab))
-
-      +header('events', 'Events')
-      div.apis.panel-collapse.collapse#palette-tab-events
-
-      +header('handlers', 'Handlers')
-      div.apis.panel-collapse.collapse#palette-tab-handlers
-
-      +header('properties', 'Properties')
-      div.apis.panel-collapse.collapse#palette-tab-properties
-
-      +header('snippets', 'Snippets')
-      div.apis.panel-collapse.collapse#palette-tab-snippets
+      if entryGroups
+        each entries, group in entryGroups
+          +header(group, group)
+          div.apis.panel-collapse(id="palette-tab-" + _.string.slugify(group))
```

Comment:
Regarding using `_.string.slugify` method. This method is in some ways a problematic method as it isn't a lodash method. It's a method we have patched onto the global lodash object.
Thus this line of code makes us more coupled to the global version 2 of lodash on the site. If we moved the `slugify` logic to a `js` file we could use the `underscore.string` library directly.

One day when trying to move away from, or while updating lodash, these lines will crop up. If it is difficult to refactor this logic into `js` then this tradeoff is probably worth it for now.

What do you think?

Reply:
Alternatively you could import 'underscore.string' in the SpellPaletteView.coffee file and use the slug method directly.

Reply:
Thanks for the great suggestions, I agree that this might be problematic when we try to upgrade/remove lodash.
Since this is used at a lot of places in the legacy code(specially PlayLevelView and related files), this has been propagated into the forked PlayLevelView for ozaria.

I like your suggestion of moving this to a common js file, however, I am not sure about the naming and location for that file. I think I will keep it as it is for now since this is coming from the legacy code and will be mindful and think about it a bit more in the future when writing new code that involves slugifying.

Thanks for pointing this out!